notebooks/02-ingestion/05-partitioning-data.ipynb

```diff
@@ -0,0 +1,717 @@
+    "Let's get one more data point...\n",
+    "\n",
+    "#### Concurrency = 8"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "ec00d899-7f25-4254-ab23-b79a7920e9e3",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "use_table=\"flights_year_50k\"\n",
+    "print (f\"{use_table} {test_in_parallel(sql.format(use_table), 8, 50)}\")\n",
+    "\n",
+    "use_table=\"flights_year_IATA\"\n",
+    "print (f\"{use_table} {test_in_parallel(sql.format(use_table), 8, 50)}\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "f36ff06e-e2ab-4a7a-8c08-c9966b336347",
+   "metadata": {
+    "jp-MarkdownHeadingCollapsed": true
+   },
+   "source": [
+    "My results:\n",
+    "```\n",
+    "flights_year_50k Results  = count:400  avg:25.969 ms   min:11.595 ms  max:97.475 ms\n",
+    "flights_year_IATA Results = count:400  avg:24.171 ms   min:11.979 ms  max:43.507 ms\n",
+    "```\n",
+    "Again, the avg continues to grow and the minimum continues unchanged.\n",
+    "\n",
+    "The important difference continues to be in the max value that is now `2.25` x faster on the IATA table.\n",
+    "\n",
+    "As we increase concurrency the IATA table continues to be more effective at dealing with concurrency. This effect will be more noticeable in a clustered deployment with more independent resources and as segments sizes grow to their more normal 3-10 million rows each."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "44738d6d-cec2-40ad-aaba-998c758c63f4",
+   "metadata": {},
+   "source": [
+    "## Clean up\n",
+    "\n",
+    "Run the following cell to remove the tables created in the database."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "8082b545-ba7f-4ede-bb6e-2a6dd62ba0d8",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "druid.datasources.drop(\"flights_hour\")\n",
+    "druid.datasources.drop(\"flights_year\")\n",
+    "druid.datasources.drop(\"flights_year_50k\")\n",
+    "druid.datasources.drop(\"flights_year_IATA\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "54b8d5fe-ba85-4b5b-9669-0dd47dfbccd1",
+   "metadata": {},
+   "source": [
+    "## Summary\n",
+    "\n",
+    "* You learned that\n",
+    "  * Druid works best with fewer optimally sized segments.\n",
+    "  * Optimal segment size will vary based on the use case, but 5 million rows is a good starting point.\n",
+    "  * PARTITIONED BY is used to define the segment granularity of a table.\n",
+    "  * CLUSTERED BY is used to organize segments within a segment granularity time interval.\n",
+    "  * CLUSTERED BY columns should be selected based on the most common filter criteria for the queries it will serve.\n",
+    "  \n",
+    "## Learn more\n",
+    "\n",
+    "* Take a look at the [06-partitioning-while-streaming.ipynb](06-partitioning-while-streaming.ipynb) notebook which takes a look at how to address partitioning and clustering for streaming.\n",
```

Comment:
Are there any amendments people could make to this notebook themselves to try different things out?

Reply:
Sure. That's the point of leaving the tables with all the columns from the source. 
They can test with different columns for secondary partitioning, test other segment granularities, etc...